cmd/influx/main.go

```diff
@@ -160,18 +174,102 @@ func main() {
 				Usage: "Check the InfluxDB /health endpoint",
 				Flags: commonFlags,
 				Action: func(ctx *cli.Context) error {
-					client, err := newApiClient(ctx, false)
+					cli, err := newCli(ctx)
+					if err != nil {
+						return err
+					}
+					client, err := newApiClient(ctx, cli, false)
+					if err != nil {
+						return err
+					}
+					return cli.Ping(ctx.Context, client.HealthApi)
+				},
+			},
+			{
+				Name:  "setup",
+				Usage: "Setup instance with initial user, org, bucket",
+				Flags: append(
+					commonFlagsNoToken,
+					&cli.StringFlag{
+						Name:    "username",
+						Usage:   "Name of initial user to create",
+						Aliases: []string{"u"},
+					},
+					&cli.StringFlag{
+						Name:    "password",
+						Usage:   "Password to set on initial user",
+						Aliases: []string{"p"},
+					},
+					&cli.StringFlag{
+						Name:        tokenFlag,
+						Usage:       "Auth token to set on the initial user",
+						Aliases:     []string{"t"},
+						EnvVars:     []string{"INFLUX_TOKEN"},
+						DefaultText: "auto-generated",
+					},
+					&cli.StringFlag{
+						Name:    "org",
+						Usage:   "Name of initial organization to create",
+						Aliases: []string{"o"},
+					},
+					&cli.StringFlag{
+						Name:    "bucket",
+						Usage:   "Name of initial bucket to create",
+						Aliases: []string{"b"},
+					},
+					&cli.StringFlag{
+						Name:        "retention",
+						Usage:       "Duration initial bucket will retain data, or 0 for infinite",
+						Aliases:     []string{"r"},
+						DefaultText: "infinite",
+					},
+					&cli.BoolFlag{
+						Name:    "force",
+						Usage:   "Skip confirmation prompt",
+						Aliases: []string{"f"},
+					},
+					&cli.StringFlag{
+						Name:    "name",
+						Usage:   "Name to set on CLI config generated for the InfluxDB instance, required if other configs exist",
+						Aliases: []string{"n"},
+					},
+					&cli.BoolFlag{
+						Name:    printJsonFlag,
+						Usage:   "Output data as JSON",
+						EnvVars: []string{"INFLUX_OUTPUT_JSON"},
+					},
+					&cli.BoolFlag{
+						Name:    hideHeadersFlag,
+						Usage:   "Hide the table headers in output data",
+						EnvVars: []string{"INFLUX_HIDE_HEADERS"},
+					},
```

Comment:
Would it be useful to separate this list of flags out as a separate variable like `commonFlags` and `commonFlagsNoToken` are their own variables? Maybe not, since I'd imagine `commonFlags` and `commonFlagsNoToken` will be used in many other commands, but just a thought.

Reply:
`printJsonFlag` and `hideHeadersFlag` can/should be factored out eventually, as they're common to many commands. The rest I'd like to keep as-is because `setup` has unique descriptions for each, and IMO it's nice to have minimal distance between where the flags are defined and where they're looked up in the `Action` below.